lib/matplotlib/axes/_axes.py

```diff
@@ -6407,34 +6407,11 @@ def _normalize_input(inp, ename='input'):
 
             # adopted from adjust_x/ylim part of the bar method
             if orientation == 'horizontal':
-                xmin0 = max(_saved_bounds[0]*0.9, minimum)
                 xmax = self.dataLim.intervalx[1]
-                for m in n:
-                    # make sure there are counts
-                    if np.sum(m) > 0:
-                        # filter out the 0 height bins
-                        xmin = np.amin(m[m != 0])
-                    # If no counts, set min to zero
-                    else:
-                        xmin = 0.0
-                xmin = max(xmin*0.9, minimum) if not input_empty else minimum
-                xmin = min(xmin0, xmin)
-                self.dataLim.intervalx = (xmin, xmax)
+                self.dataLim.intervalx = (0, xmax)
```

Comment:
The original code was broken, even before 2.x; the present code is broken.  It turns out that the present "sticky" algorithm is not doing the right thing for plain bar plots, either, when switching to a log scale; this was working in 1.5.3.  I don't know if it was working with the "margins" algorithm in 2.x.